### DISCUSSION TITLE
Project classic future

### DISCUSSION THREAD

**Discussion by soundstep** [community/community#124801] [Projects and Issues] [answered]
### Select Topic Area

Question

### Feature Area

Projects

### Body

Hi,

We have created a workflow involving:
- A Github classic project in a repo
- Github API around projects and cards
- Github Action workflows
- Github Repo Discussions

The workflow is intricate, we heavily rely on the Github API, for example, to find out when people are moving cards between columns. This was impossible at the time of creation using the new projects, as the API was simply not there.

We have seen a today a prompt appear:
<img width="245" alt="Screenshot 2024-05-20 at 11 33 04" src="https://github.com/community/community/assets/133293/a9429fa0-b6d5-4341-a99d-da07f3fe5295">

This probably would break the workflow (we'll investigate).

I was also unable to remove this prompt in the settings by disabling the new projects:
<img width="808" alt="Screenshot 2024-05-20 at 11 23 58" src="https://github.com/community/community/assets/133293/f30d32f7-ee6d-4d07-8cab-564f7606733e">

A few questions:

1. Can we disable this prompt in a repo?
2. What is the future of the project classic?
3. Are we going to be forced into migrating?

Cheers.

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by xmo-odoo** [comment]
Apparently github has also managed to break drag and drop of cards in classic projects.

**Reply by evi-liu** [reply]
Could you send a screenshot or gif of the bug you're experiencing?

**Reply by xmo-odoo** [reply]
Looks like it works again, sorry about the false alarm, might have been some sort of caching issue in the browser (firefox FDE).

Basically when I posted that message trying to drag a card would select it but the drag/drop would not fire at all, the cards would not move.

**Comment by evi-liu** [comment, marked as answer]
:wave: We recently announced that [Projects Classic will be sunset](https://github.blog/changelog/2024-05-23-sunset-notice-projects-classic/#github-com-timeline). We would recommend that you migrate your classic projects to the new projects. This will mean that instead of using the [REST API for Projects classic](https://docs.github.com/en/rest/projects?apiVersion=2022-11-28), you can instead use the [Projects GraphQL API](https://docs.github.com/en/issues/planning-and-tracking-with-projects/automating-your-project/using-the-api-to-manage-projects) in your workflows.

If you do encounter any difficult when using our Projects GraphQL, please do let us know so we can take a look and address any gaps from our end.

**Comment by soundstep** [comment]
@evi-liu Am I right to say that Github permissions around the "new project" (are they called org projects?) for being able to make API requests with tokens generated from Github Apps, are falling into the "org permissions" section of those Github Apps?

This means that I, as a Github App manager, will now require an org admin to update any update to a Github App (repo changes and permission changes, apparently even for repo permissions if there is at least 1 org permission)? Basically, being or not being a Github App Manager will make no difference if I add org project permissions?

<img width="756" alt="Screenshot 2024-05-30 at 12 17 31" src="https://github.com/community/community/assets/133293/45a38c01-6e22-4d38-895f-1d49ad6f79e5">

For repo permissions:
<img width="762" alt="Screenshot 2024-05-30 at 11 40 51" src="https://github.com/community/community/assets/133293/e49d134a-accb-4f4d-bd90-5cc50b9edc01">

For org permissions:
<img width="770" alt="Screenshot 2024-05-30 at 11 41 46" src="https://github.com/community/community/assets/133293/a258d724-a679-4ada-ab50-7d21a178f732">

**Reply by evi-liu** [reply]
Hi @soundstep, if you're a GitHub App manager, you should be able to manage apps owned by the organization ([docs reference](https://docs.github.com/en/apps/maintaining-github-apps/about-github-app-managers)).  

Is this not working for you? If so, we can take a look into this and help resolve this.

**Comment by soundstep** [comment]
@evi-liu I can manage a Gihub App and change permissions, but for any app that has at least one org permission, I cannot update the installation of this app without an org owner. This probably has always been the case, but moving classic project permission from a repo scope to an org scope for the new project has changed what I can and can't do.

Let me demonstrate that. The following screenshots and changes have been made by myself on 2 Github App while being a Github App Manager:

- `fe-core-bot` (only has repo permissions)
- `fe-reveal-bot` (has repo permissions and one org permission: the new project)

**Summary for the following**: migrating a project classic to an org project has indirectly removed my capability to update a Github App installation as a Github App Manager.

## `fe-core-bot`: Github App with repo-only permissions

1. `Install App` navigation item is present:

<img width="1159" alt="Screenshot 2024-06-03 at 08 27 25" src="https://github.com/community/community/assets/133293/74603867-33e3-4706-bc4e-7ee7ba78e69e">

2. Changing permission with a repo scope works and keeps the installation in the navigation:

<img width="549" alt="Screenshot 2024-06-03 at 08 28 15" src="https://github.com/community/community/assets/133293/8abaeae6-f7c8-4d7b-b380-9a8abeabe907">

3. Updating the installation works:

<img width="1456" alt="Screenshot 2024-06-03 at 08 28 40" src="https://github.com/community/community/assets/133293/c2f5b5fc-d3f7-4061-ac51-969158edbd38">

## `fe-reveal-bot`: Github App with at least one org permission

1. The `Install App` navigation item is NOT present (this is the first difference and see that it has one org permission):

<img width="1172" alt="Screenshot 2024-06-03 at 08 27 00" src="https://github.com/community/community/assets/133293/5ba71eed-4121-49fd-8f7f-fdade237de2d">

2. Changing permission with a repo scope, while it has at least one org permission, works and the `Install App` menu item is still missing in the navigation:

<img width="562" alt="Screenshot 2024-06-03 at 08 29 06" src="https://github.com/community/community/assets/133293/68361f5e-5093-45c8-874b-466b4011a18a">

3. Updating the installation is NOT possible anymore. This happens even if only a repo permission has been changed, it only need to have one (changed or unchanged) org permission to prevent a Github App manager from updating the installation. The permission added in the following screenshot is a repo scope "codespace" permission from "no access" to "read only":

<img width="1191" alt="Screenshot 2024-06-03 at 08 29 47" src="https://github.com/community/community/assets/133293/f0e9d695-f30f-4c9d-87e0-eb8f2fc546a1">

**Reply by basic99** [reply]
I tried the new projects and found it to be clunky. The classics project was simple (supposed to be like post-it notes?) and I liked it and sorry to see it going.

**Reply by evi-liu** [reply]
@soundstep Got it, and thank you for walking the difference between the two flows. Unfortunately, managing installations for organization-level projects without an organization owner is currently not possible.

I understand that this may feel like an unnecessary obstacle, especially considering that you could do this without organization owners in Projects Classic. We will keep monitoring feedback to see if this is something we should consider adjusting in the future.

**Reply by soundstep** [reply]
No worries @evi-liu. I expected that only changing an org permission would require an org admin, not when changing repo permission without changing any org permission. I get this is a "niche" issue, thanks for looking and let us know if changes are made in a near future.
